cmd/sidecars/network-passt-binding/domain/configurator.go

```diff
@@ -95,6 +95,25 @@ func (p PasstNetworkConfigurator) Mutate(domainSpec *domainschema.DomainSpec) (*
 	} else {
 		domainSpecCopy.Devices.Interfaces = append(domainSpecCopy.Devices.Interfaces, *generatedIface)
 	}
+	const (
+		sharedMemoryBackingAccessMode = "shared"
+		memfdMemoryBackingSourceType  = "memfd"
+	)
+	if domainSpecCopy.MemoryBacking == nil {
+		domainSpecCopy.MemoryBacking = &domainschema.MemoryBacking{
+			Access: &domainschema.MemoryBackingAccess{
+				Mode: sharedMemoryBackingAccessMode,
+			},
+			Source: &domainschema.MemoryBackingSource{
+				Type: memfdMemoryBackingSourceType,
+			},
+		}
+	} else {
+		if domainSpecCopy.MemoryBacking.Access != nil && domainSpecCopy.MemoryBacking.Access.Mode != sharedMemoryBackingAccessMode {
+			return nil, fmt.Errorf("memory backing access mode must be 'shared'; cannot override existing mode: %s",
+				*domainSpecCopy.MemoryBacking.Access)
```

Comment:
Since this condition is critical and there is not need to continue if it is met - I suggest moving it to the top and return early.

Additionally, there is an implicit assumption that all current and future source types are valid.
How about we limit the source configuration to something we know would work?

Reply:
@sbrivio-rh, could you please share your opinion on this one?

> Additionally, there is an implicit assumption that all current and future source types are valid.
How about we limit the source configuration to something we know would work?

Reply:
Both `shared` and `memfd` work with libvirt plus passt.

I didn't test things with KubeVirt. But yes, ideally, all the options should work because where the memory comes from / how it's mapped _should_ be independent from how it's used. So the assumption looks correct to me.

If somebody finds a configuration that doesn't work, I think it's good that KubeVirt tries to apply that in the first place, so that the bug report will have meaningful details. But this is a matter of project policies I suppose.

Reply:
Thank you for clarifying @sbrivio-rh!